src/dfx/src/main.rs

```diff
@@ -201,7 +201,10 @@ fn main() {
                 Err(e) => Err(e),
             }
         }
-        Err(e) => Err(e),
+        Err(e) => match command {
+            commands::Command::Schema(_) => commands::exec_without_env(command),
+            _ => Err(e),
+        },
```

Comment:
If the failure of the `schema` command can't depend on the failure to construct an Environment, what's the purpose of displaying the earlier error?
```suggestion
        Err(_) => match command {
            commands::Command::Schema(_) => commands::exec_without_env(command),
            Err(e) => Err(e),
        },
```

Reply:
I think you're misreading the code (meaning I have to clarify it somehow). The match doesn't match on the result of the schema command -  it figures out if the command being run can be executed without having an Environment available.

A different way of writing a similar thing would be: 
```suggestion
        Err(original_error) => commands::exec_without_env(command).or_else( || Err(original_error)),
        },
```